plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java

```diff
@@ -72,7 +70,7 @@ public static PreallocationType getPreallocationType(final Storage.ProvisioningT
             case FAT:
                 return PreallocationType.Full;
             default:
-                throw new NotImplementedException();
+                throw new NotImplementedProvisioningTypeException();
```

Comment:
What if instead, you throw a cloud runtime with a message describing the problem/error here?

Reply:
That is certainly an option, @rafaelweingartner . I had discussed with @nvazquez and he suggested the commons-lang NotImplementedException as it is loaded through the storage module anyhow. Thoughts?

Reply:
It also works. I only think that we need to add something in the exception. I mean, a message saying what is not implemented (the preallocation type selected/sent by the user).

Reply:
check

Reply:
Agree :+1: